src/code_80005FD0.c

```diff
@@ -1309,9 +1309,9 @@ s32 func_80008E58(s32 payerId, s32 pathIndex) {
     posY = player->pos[1];
     posZ = player->pos[2];
     stackPadding = pathIndex;
-    trackSegment = func_802ABD40(player->unk_110.unk3A);
+    trackSegment = get_track_segment(player->unk_110.collisionMeshIndex);
     D_80163318[payerId] = trackSegment;
-    D_80162FCE = func_8000C0BC(posX, posY, posZ, trackSegment, &pathIndex);
+    D_80162FCE = find_closest_waypoint_track_segment(posX, posY, posZ, trackSegment, &pathIndex);
```

Comment:
Not sure I like the _track_segment part.

Should use trackSection instead of trackSegment I think.